<a href="https://colab.research.google.com/github/imaleaf03/bt9/blob/main/06_house.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [3]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows:
1. The first a random number from 1 to 4
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [4]:
#np.random.seed(42)
series1 = pd.Series(np.random.randint(1, 5, size=100))
series2 = pd.Series(np.random.randint(1, 4, size=100))
series3 = pd.Series(np.random.randint(10000, 30001, size=100))

print("Series 1:")
print(series1.head())

print("\nSeries 2:")
print(series2.head())

print("\nSeries 3:")
print(series3.head())

Series 1:
0    1
1    2
2    4
3    3
4    3
dtype: int64

Series 2:
0    2
1    3
2    3
3    2
4    3
dtype: int64

Series 3:
0    15370
1    19241
2    25845
3    17924
4    17133
dtype: int64


### Step 3. Let's create a DataFrame by joinning the Series by column

In [5]:
df = pd.concat([series1, series2, series3], axis=1)
df

0  1      2
0   1  2  15370
1   2  3  19241
2   4  3  25845
3   3  2  17924
4   3  3  17133
.. .. ..    ...
95  1  3  11536
96  1  2  18864
97  3  1  14488
98  3  3  12665
99  1  1  22435

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [9]:
df.rename(columns={0: 'bedrs', 1: 'bathrs', 2: 'price_sqr_meter'})

bedrs  bathrs  price_sqr_meter
0       1       2            15370
1       2       3            19241
2       4       3            25845
3       3       2            17924
4       3       3            17133
..    ...     ...              ...
95      1       3            11536
96      1       2            18864
97      3       1            14488
98      3       3            12665
99      1       1            22435

[100 rows x 3 columns]

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [18]:
bigcolumn = pd.concat([series1, series2, series3], ignore_index=True)
bigcolumn

0          1
1          2
2          4
3          3
4          3
       ...  
295    11536
296    18864
297    14488
298    12665
299    22435
Length: 300, dtype: int64

### Step 6. Oops, it seems it is going only until index 99. Is it true?

### Step 7. Reindex the DataFrame so it goes from 0 to 299